src/praisonai-agents/praisonaiagents/skills/remote.py

```
@@ -0,0 +1,216 @@
+        base = self._cache_base(cache_dir)
+        base.mkdir(parents=True, exist_ok=True)
+
+        tmp_clone: Optional[str] = None
+        try:
+            tmp_clone = tempfile.mkdtemp(prefix="praison-skill-clone-")
+            clone_path = Path(tmp_clone) / "repo"
+            cmd = ["git", "clone", "--depth=1"]
+            if self.ref:
+                cmd += ["--branch", self.ref]
+            cmd += [self.url, str(clone_path)]
+            proc = subprocess.run(cmd, capture_output=True, text=True)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)